indico/modules/events/abstracts/util.py

```diff
@@ -379,3 +382,15 @@ def get_events_with_abstract_persons(user, dt=None):
     for person in query:
         data[person.event_id].add('abstract_person')
     return data
+
+
+def filter_field_values(fields, user, managers, submitter):
+    if user in managers:
+        return {field for field in fields if field.contribution_field.is_active}
+    if user == submitter:
+        return {field for field in fields
```

Comment:
I would pre-filter the fields based on `is_active` before the final set comprehensions. More readable that way.